### PR TITLE
Customize PR instructions based on files modified

### LINKED ISSUE #5182: Customize PR instructions based on files in the PR
### Dependency
- [x] #5165 

### Overview
We need to customize the pr instructions message.  If a PR contains multiple files including CONTRIBUTING.md, the current pr instructions are appropriate.  If CONTRIBUTING.md is the only file modified, the instructions for pulling down the issue branch locally can be omitted, and if CONTRIBUTING.md is not  modified, the instruction to preview CONTRIBUTING.md on GitHub can be omitted.  

### Action Items
- [x] Become familiar with the resources describing the HfLA GHA archtecture[^1] and testing procedures[^2] as well as the resource describing the PR Instructions GHA[^3]
- [x] Update create-instruction.js[^7] so that it examines the files changed in the PR.  
- [x] Change the implementation so that the special instruction for reviewing CONTRIBUTING.md is provided only if the PR includes a change to CONTRIBUTING.md.  
- [x] If the only file changed in the PR is CONTRIBUTING.md, then do not display the PR instruction that provides git commands pulling down the issue branch locally.  

### Resources/Instructions
[^1]:[Hack for LA's GitHub Actions](https://github.com/hackforla/website/wiki/Hack-for-LA's-GitHub-Actions)
[^2]:[HfLA GitHub Actions](https://github.com/hackforla/website/wiki/HfLA-GitHub-Actions)
[^3]:[GHA: pr instructions](https://github.com/hackforla/website/wiki/GHA:-pr-instructions)
[^4]:[PR Instructions Template](https://github.com/hackforla/website/blob/gh-pages/github-actions/pr-instructions/pr-instructions-template.md)
[^5]:[PR Instructions Workflow](https://github.com/hackforla/website/blob/gh-pages/github-actions/pr-instructions/post-comment.js)
[^6]:[WR PR Instructions Workflow](https://github.com/hackforla/website/blob/gh-pages/.github/workflows/wr-pr-instructions.yml)
[^7]: https://github.com/hackforla/website/blob/gh-pages/github-actions/pr-instructions/create-instruction.js
[^8]:https://github.com/hackforla/website/blob/gh-pages/github-actions/pr-instructions/post-comment.js

### PR DESCRIPTION
**Pull request by anthonypz**
Fixes #5182

### What changes did you make?
  - This change grabs a list of modified files submitted in the Pull Request (using the global `github`  and `context` variables).
  - Wrote logic to only include the Pull Request and Contributor instructions based on what files are modified in the PR.

### Why did you make the changes (we will use this info to test)?
  - The changes were made so that PR instructions not relevant to the Pull Request are omitted, eliminating confusion for contributors.

### CODE REVIEW

File: github-actions/pr-instructions/create-instruction.js [hunk 1]
```diff
@@ -53,7 +53,7 @@ async function getModifiedFiles() {
     const ownerName = context.payload.pull_request.head.repo.owner.login;
 
     // Gets the list of files modified in the pull request and destructures the data object into a files variable
-    const { data: files } = await github.pulls.listFiles({
+    const { data: files } = await github.rest.pulls.listFiles({
```
Comment (anthonypz) [inline github-actions/pr-instructions/create-instruction.js:62]: octokitReference
According to the [Octokit docs](https://octokit.github.io/rest.js/v20), the `rest` property is needed when calling the API. This change fixes my previous commit where a test failed due to a type error.

**Review by t-will-gillis — CHANGES_REQUESTED** [review]
Hey @anthonypz Good job with this PR so far. I can follow your logic and understand how this should be working; however, when I test it in my repo the message is not being posted. 

Here is one [test issue](https://github.com/t-will-gillis/website/pull/627) I ran in my repo. As can be seen, the bot is posting `[object Promise]` instead of the message. I believe that the code is mostly working but something is breaking down.

Since this issue directly relates to a GitHub Action, this issue really needs to be tested as a GHA which means you will need to set up the testing environment in your own repo to demonstrate the GHA works. [This sheet](https://github.com/hackforla/website/issues/6537#issuecomment-2041147335) is probably the best starting point as well as the documents listed on #5182. 

I can help you once you get set up- feel free to post questions on this issue and on Slack. 

Thanks for working on this- as I said, I think the code is mostly there.

**Comment by t-will-gillis** [comment]
Hey @anthonypz  Great job! I can confirm this is working now: without `CONTRIBUTING.md` the [PR review message](https://github.com/t-will-gillis/website/pull/629) is the same as previous, with only `CONTRIBUTING.md` the [PR review message](https://github.com/t-will-gillis/website/pull/630) only refers to that file, and with `CONTRIBUTING.md` and another file both [PR messages](https://github.com/t-will-gillis/website/pull/631) are shown. 


I would like to run a couple of formatting questions by @roslynwythe that are beyond the scope of the original issue. I think that the message for `CONTRIBUTING.md` only is a little awkward with the heavy line at the beginning and with the omission of the "Want to review this pull request..." message:
<details>
Screenshot 2024-04-24 152335

</details>  

- I believe that for any message, we would like to include the following, initial statement. To accomplish this, the statement would either be a third .md file in the `/pr-instructions/` folder or possibly(?) hardcoded into `create-instructions.js`:

>Want to review this pull request? Take a look at [this documentation](https://github.com/hackforla/website/wiki/How-to-Review-Pull-Requests) for a step by step guide.

- Then I would opt for either removing the `----------------` from line 3 of `pr-instructions-contrib-template.md` to remove the heavy line completely, or alternately inserting <blankline> + `-----------` + <blankline> before line 5 of `pr-instructions-template.md` so those both messages include the heavy separation. (Note that `---------------` can be `---`)

**Comment by anthonypz** [comment]
Hey @t-will-gillis, thanks for reviewing my changes. Good catch - I agree that your suggested changes make more sense. I'd be happy to implement them as part of this pull request as soon as we know what those changes should look like.

**Comment by roslynwythe** [comment]
@t-will-gillis I agree that your suggested changes are worthwhile.  Thank you

**Comment by t-will-gillis** [comment]
Hey @anthonypz Let's go ahead with this then. In the end the comments for the three situations (with `CONTRIBUTING.md` and another file(s), with only `CONTRIBUTING.md`, and with only other files) would look similar to [this demo issue](https://github.com/t-will-gillis/website/issues/632).

--> Feel free to make suggestions if you see something...

To confirm:

- The following message should appear at the top for all situations
  - For consistency, this could be in a file named `pr-instructions-header.md` 
  - This is definitely out of scope and could be done in a future issue, if ever: _If you wanted to figure out the logic, I think it would be possible to have all three messages in the same .md file._ 

```
Want to review this pull request? Take a look at [this documentation](https://github.com/hackforla/website/wiki/How-to-Review-Pull-Requests) for a step by step guide!  
```  
- `pr-instructions-template.md` 
  - would no longer have the above statement
  - would add `---` between lines 2 and 5 similar to `pr-instructions-contrib-template.md`
- `pr-instructions-contrib-template.md`
  - essentially ok, could replace line 3 with `---`

Logic in `create-instructions.js` as needed to create the appropriate messages.

Thank you for doing this work beyond the scope of the original issue!

**Comment by anthonypz** [comment]
Thanks for the detailed instructions and providing a demo issue, @t-will-gillis. It was very helpful. I went ahead and made the requested changes, which I'm testing now.

**Review by t-will-gillis — CHANGES_REQUESTED** [review]
Hey @anthonypz Great work! I tested for the three different files scenarios, and the appropriate PR review message is posting on the issue. It looks like you were able to get a testing environment set up on your repo also. Looks good and thanks for making the extra revisions.

This is ready to merge, but I have one more request. It looks like in `create-instructions.js` lines 32, 50, 52, and 53 the indent from the original file is unaligned (tabs used instead of spaces?) Would you please fix this?

Great job again!

**Comment by anthonypz** [comment]
@t-will-gillis That's strange, VS Code usually converts tabs to spaces for me. Not sure what happened there.

I've just pushed a fix. Please let me know if anything else is needed.

**Review by t-will-gillis — APPROVED** [review]
I am not even sure how to tab in VSCode because it converts for me also so maybe someone earlier was using a different editor?

Perfect- thanks again

**Comment by t-will-gillis** [comment]
Hey @anthonypz  We had to revert this PR because the workflow had an error. I left a message for you on Slack also.

**Pull request merged**